### ISSUE TITLE
bun 1.1.41 it bug, 1.1.40 it still normal

### ISSUE THREAD

**Issue by 0xc0000f** [oven-sh/bun#15913]
### How can we reproduce the crash?

```ts
#10 0.060 bun install v1.1.41 (b8f28ed8)
#10 0.063 Resolving dependencies
#10 0.835 Resolved, downloaded and extracted [2]
#10 2.036 ============================================================
#10 2.036 Bun v1.1.41 (b8f28ed8) Linux x64 (baseline)
#10 2.036 Linux Kernel v6.8.0 | glibc v2.31
#10 2.036 CPU: sse42 popcnt avx avx2
#10 2.036 Args: "node" "scripts/postinstall.js"
#10 2.036 Features: jsc
#10 2.036 Builtins: "bun:main" "node:fs" "node:path" "node:string_decoder" "node:util/types"
#10 2.036 Elapsed: 56ms | User: 44ms | Sys: 15ms
#10 2.036 RSS: 1.07GB | Peak: 0.44GB | Commit: 1.07GB | Faults: 0
#10 2.036
#10 2.036 panic(main thread): Segmentation fault at address 0x0
#10 2.036 Crashed while visiting /temp/dev/node_modules/@prisma/get-platform/dist/chunk-6HZWON4S.js
#10 2.036 oh no: Bun has crashed. This indicates a bug in Bun, not your code.
#10 2.036
#10 2.036 To send a redacted crash report to Bun's team,
#10 2.036 please file a GitHub issue using the link below:
#10 2.036
#10 2.036 https://bun.report/1.1.41/Bn1b8f28edAgggEog9hlE+8mO0qz3nDig49hD2i2rhD++6wgD0l9vgDujqxgD4+4k9D_A2AA
#10 2.036
#10 2.036
#10 2.036 error: postinstall script from "@prisma/engines" terminated by SIGILL (Illegal instruction)
#10 2.218 Illegal instruction (core dumped)
#10 ERROR: process "/bin/sh -c cd /temp/dev && bun install --frozen-lockfile" did not complete successfully: exit code: 132
------
> [install 3/6] RUN cd /temp/dev && bun install --frozen-lockfile:
2.036 oh no: Bun has crashed. This indicates a bug in Bun, not your code.
2.036
2.036 To send a redacted crash report to Bun's team,
2.036 please file a GitHub issue using the link below:
2.036
2.036 https://bun.report/1.1.41/Bn1b8f28edAgggEog9hlE+8mO0qz3nDig49hD2i2rhD++6wgD0l9vgDujqxgD4+4k9D_A2AA
2.036
2.036
2.036 error: postinstall script from "@prisma/engines" terminated by SIGILL (Illegal instruction)
2.218 Illegal instruction (core dumped)
------
Dockerfile:8
--------------------
6 | RUN mkdir -p /temp/dev
7 | COPY package.json bun.lockb /temp/dev/
8 | >>> RUN cd /temp/dev && bun install --frozen-lockfile
9 |
10 | RUN mkdir -p /temp/prod
--------------------
ERROR: failed to solve: process "/bin/sh -c cd /temp/dev && bun install --frozen-lockfile" did not complete successfully: exit code: 132
[2024-Dec-20 22:11:22.212568] Deployment failed. Removing the new version of your application.
```

### Relevant log output

```shell

```

### Stack Trace (bun.report)

Bun v1.1.41 ([`b8f28ed`](<https://github.com/oven-sh/bun/tree/b8f28ed8afd1c2b60568b2b0158d39a674178027>)) on linux x86_64_baseline [RunAsNodeCommand]

Segmentation fault at address 0x00000000

- [`js_parser.zig:4272`](<https://github.com/oven-sh/bun/blob/b8f28ed8afd1c2b60568b2b0158d39a674178027/src/js_parser.zig#L4272>): `src.js_parser.Parser._parse__anon_141505`
- [`js_parser.zig:3098`](<https://github.com/oven-sh/bun/blob/b8f28ed8afd1c2b60568b2b0158d39a674178027/src/js_parser.zig#L3098>): `src.cache.JavaScript.parse`
- [`bundler.zig:1408`](<https://github.com/oven-sh/bun/blob/b8f28ed8afd1c2b60568b2b0158d39a674178027/src/bundler.zig#L1408>): `src.bundler.Bundler.parseMaybeReturnFileOnlyAllowSharedBuffer__anon_121976`
- [`bundler.zig:1255`](<https://github.com/oven-sh/bun/blob/b8f28ed8afd1c2b60568b2b0158d39a674178027/src/bundler.zig#L1255>): `src.bundler.Bundler.parseMaybeReturnFileOnly__anon_117118`
- [`module_loader.zig:1656`](<https://github.com/oven-sh/bun/blob/b8f28ed8afd1c2b60568b2b0158d39a674178027/src/bun.js/module_loader.zig#L1656>): `src.bun.js.module_loader.ModuleLoader.transpileSourceCode__anon_117062`
- [`module_loader.zig:2367`](<https://github.com/oven-sh/bun/blob/b8f28ed8afd1c2b60568b2b0158d39a674178027/src/bun.js/module_loader.zig#L2367>): `Bun__transpileFile`
- [`ModuleLoader.cpp:609`](<https://github.com/oven-sh/bun/blob/b8f28ed8afd1c2b60568b2b0158d39a674178027/src/bun.js/bindings/ModuleLoader.cpp#L609>): `Bun::jsFunctionRequireCommonJS`

Features: jsc

<sub>Sentry Issue: <strong><a href="https://bun-p9.sentry.io/issues/6164820105/">BUN-A3S</a></strong></sub>

**Comment by 0xc0000f** [comment]
![Image](https://github.com/user-attachments/assets/3bbd01f8-4048-424d-aefc-c423b92194d8)
1.1.40 it working fine

**Comment by Jarred-Sumner** [comment]
Fixed by @dylan-conway in #15911
